app/services/catalog/approval_transition.rb

```diff
@@ -20,30 +20,36 @@ def process
 
     def state_transitions
       if approved?
-        submit_order
         @state = "approved"
+        submit_order
       elsif denied?
-        mark_denied
         @state = "denied"
+        mark_denied
       else
         @state = "pending"
+        @order_item.order.transition_state
       end
     end
 
     def submit_order
       @order_item.update_message("info", "Submitting Order #{@order_item.order_id} for provisioning ")
       Catalog::SubmitOrder.new(@order_item.order_id).process
+      finalize_order("Approved")
     rescue Catalog::TopologyError => e
       Rails.logger.error("Error Submitting Order #{@order_item.order_id}, #{e.message}")
       @order_item.update_message("error", "Error Submitting Order #{@order_item.order_id}, #{e.message}")
     end
 
     def mark_denied
-      @order_item.update!(:state => "Denied")
-      @order_item.order.update!(:state => "Denied")
+      finalize_order("Denied")
       @order_item.update_message("info", "Order #{@order_item.order_id} has been denied")
     end
 
+    def finalize_order(state)
```

Comment:
Do we even need to pass in the `state` here? It's an `attr_reader` already, so we should just be able to use it.

Reply:
Good point! Updated to use the instance variable instead of a parameter.